src/node/blockstorage.cpp

```diff
@@ -601,7 +601,16 @@ bool BlockManager::FindBlockPos(FlatFilePos& pos, unsigned int nAddSize, unsigne
 
     bool finalize_undo = false;
     if (!fKnown) {
-        while (m_blockfile_info[nFile].nSize + nAddSize >= (gArgs.GetBoolArg("-fastprune", false) ? 0x10000 /* 64kb */ : MAX_BLOCKFILE_SIZE)) {
+        unsigned int max_blockfile_size = MAX_BLOCKFILE_SIZE;
+        // Use smaller blockfiles in test-only -fastprune mode - but avoid
+        // the possibility of having a block not fit into the block file.
+        if (gArgs.GetBoolArg("-fastprune", false)) {
+            max_blockfile_size = 0x10000; // 64kiB
+            if (nAddSize >= max_blockfile_size) {
+                max_blockfile_size = nAddSize + 1;
```

Comment:
In commit "blockstorage: Adjust fastprune limit if block exceeds blockfile size" (e930814fdb9261f180d5c436cefe52e9cf38fd67)

What is the +1 for? Would be helpful to add a code comment saying what the extra byte is.

Reply:
I checked this carefully again and don't think anymore that the `+ 1` is necessary.

Reply:
I think it is necessary, as long as the while condition says
`while (m_blockfile_info[nFile].nSize + nAddSize >= max_blockfile_size)` 
it won't terminate if  `nAddSize == max_blockfile_size`, because `m_blockfile_info[nFile].nSize` will be 0 if we skip to the next block.

Maybe we could lose the `+ 1` if we'd change the `>=` from the while loop into a `>`, however I really don't want to change this critical code in this PR because it would mean we'd try to squeeze another byte into all blocks, affecting blockstorage in normal operation as well, not just the test-only `-fastprune` mode.
So I'll add a comment.

Reply:
Makes sense the `+1` is needed to be consistent with `>=` below. It does look like there is a minor bug here that will cause block files to be at most `MAX_BLOCKFILE_SIZE-1` bytes and never reach the maximum size. But if this is a bug, it goes back to 5382bcf8cd23c36a435c29080770a79b5e28af42 when this code was introduced. Agree it would not be good to change that behavior here.

Reply:
The functional test covers the `+1` nicely as well. Without the `+1` (and removing the assertion on L632) the test will hang until timed out at 30s